src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -36,11 +42,57 @@ var Marathon = React.createClass({
       deploymentsFetchState: States.STATE_LOADING,
       fetchState: States.STATE_LOADING,
       modalClass: null,
+      route: null,
       tasksFetchState: States.STATE_LOADING
     };
   },
 
   componentDidMount: function () {
+    var router = this.props.router;
+
+    router.on("route", function (route, params) {
+      this.setState({
+        route: {
+          name: route,
+          params: params
+        }
+      });
+    }.bind(this));
+
+    router.on("route:about", function () {
+      this.modalDestroy();
+      this.setState({
+        modalClass: AboutModalComponent
+      });
+    }.bind(this));
+
+    router.on("route:apps", function (appid) {
+      if (appid) {
```

Comment:
This could potentially be zero, let's use `appId != null`

I'm not sure if this will work, but I think it would be cleaner to move the `modalDestroy` out of the first `if`:
```
var app = this.state.collection ? this.state.collection.get("/" + appId) : null;
if (this.state.activeApp !== app) {
  this.modalDestroy();
}

if (app != null) {
  this.setState({
    activeApp: app,
    modalClass: AppModalComponent
  });
} else {
  this.activateTab("apps");
}